muffnn/base.py

```diff
@@ -86,80 +86,133 @@ def fit(self, X, y):
-        self.input_layer_sz_ = X.shape[1]
+            # Instantiate the graph.  TensorFlow seems easier to use by just
+            # adding to the default graph, and as_default lets you temporarily
+            # set a graph to be treated as the default graph.
+            self.graph_ = Graph()
+            with self.graph_.as_default():
+                tf_random_seed.set_random_seed(
+                    random_state.randint(0, 10000000))
 
-        # Instantiate the graph.  TensorFlow seems easier to use by just adding
-        # to the default graph, and as_default lets you temporarily set a
-        # graph to be treated as the default graph.
-        self.graph_ = Graph()
-        with self.graph_.as_default():
-            tf_random_seed.set_random_seed(random_state.randint(0, 10000000))
+                tf.get_variable_scope().set_initializer(
+                    tf.uniform_unit_scaling_initializer(self.init_scale))
 
-            tf.get_variable_scope().set_initializer(
-                tf.uniform_unit_scaling_initializer(self.init_scale))
+                self._set_up_graph()
+                self._session = tf.Session()
 
-            self._init_model()
-            self._session = tf.Session()
+                # Train model parameters.
+                self._session.run(tf.initialize_all_variables())
 
-            # Train model parameters.
-            self._session.run(tf.initialize_all_variables())
+            # Set an attributed to mark this as at least partially fitted.
+            self._fitted = True
 
+        # Train the model with the given data.
+        with self.graph_.as_default():
+            n_examples = X.shape[0]
             start_idx = 0
             epoch = 0
             indices = np.arange(n_examples)
             random_state.shuffle(indices)
 
             while True:
-                batch_ind = indices[start_idx:start_idx + batch_size]
+                batch_ind = indices[start_idx:start_idx + self.batch_size]
                 feed_dict = self._make_feed_dict(X[batch_ind],
                                                  y[batch_ind])
                 obj_val, _ = self._session.run(
                     [self._obj_func, self._train_step], feed_dict=feed_dict)
                 _LOGGER.info("objective: %.4f, epoch: %d, idx: %d",
                              obj_val, epoch, start_idx)
-                start_idx += batch_size
-                if start_idx > n_examples - batch_size:
+                start_idx += self.batch_size
+                if start_idx > n_examples - self.batch_size:
                     start_idx = 0
                     epoch += 1
                     if epoch >= self.n_epochs:
                         break
                     random_state.shuffle(indices)
 
-        # Set an attributed to make this as fitted.
-        self._fitted = True
+        return self
 
-        # Make the graph read-only.
-        self.graph_.finalize()
+    @property
+    def _is_fitted(self):
```

Comment:
i wasn't sure whether the setter was worth adding, but i'll go ahead and add it.  as you suggested, it'll probably make the code more readable.